src/utils/api/ws.test.js

```diff
@@ -0,0 +1,66 @@
+import io from 'socket.io-client';
+import ws from './ws';
+
+jest.mock('socket.io-client');
+
+describe('ws', () => {
+  const baseUrl = 'http://sample-service-url.com';
+
+  it('should return a promise and call emit with parameters', async () => {
+    const requests = [{
+      method: 'account.get',
+      params: { address: '12L' },
+    }];
+    const emit = jest.fn().mockImplementation((evtName, params, callback) => {
+      callback({ result: {} });
+    });
+    io.mockImplementation(() => ({ emit }));
+
+    const wsPromise = ws({
+      baseUrl,
+      requests,
+    });
+    await wsPromise;
+
+    expect(typeof wsPromise.then).toEqual('function');
+    expect(typeof wsPromise.catch).toEqual('function');
```

Comment:
No need to assert for these since `then` and `catch` are always functions 🤔

Reply:
Initially I wasn't testing the reopens in this test case.